sys/filedesc.c

```diff
@@ -72,9 +105,10 @@ void fdtab_unref(fdtab_t *fdt) {
    argument doest not make sense yet. */
 fdtab_t *fdtab_alloc() {
   fdtab_t *fdt = kmalloc(fd_pool, sizeof(fdtab_t), M_ZERO);
-  /* For now, fdt_files and fdt_map have a static size, so there is no need to
-   * separately allocate memory for them. */
   fdt->fdt_nfiles = NDFILE;
+  fdt->fdt_files = kmalloc(fd_pool, sizeof(file_t *) * NDFILE, M_ZERO);
+  fdt->fdt_map = kmalloc(fd_pool, bitstr_size(NDFILE), M_ZERO);
+  memset(fdt->fdt_map, 0, bitstr_size(NDFILE));
```

Comment:
You've just called `kmalloc` with `M_ZERO`, so the allocated memory is guaranteed to be cleared, no need to `memset` it. Same goes for clearing `new_fdt_map`.